src/output.F90

```diff
@@ -2162,5 +2184,239 @@ function get_label(t, i_filter) result(label)
+          lat => lattices(c % fill) % obj
+
+          select type (lat)
+
+          ! ==================================================================
+          ! RECTANGULAR LATTICES
+          type is (RectLattice)
+
+            ! Write to the geometry stack
+            path = trim(path) // "->" // to_str(lat%id)
+
+            n_x = lat % n_cells(1)
+            n_y = lat % n_cells(2)
+            n_z = lat % n_cells(3)
+
+            ! Loop over lattice coordinates
+            do m = 1, n_z
+              do l = 1, n_y
+                do k = 1, n_x
+
+                  if (final > lat % offset(map, k, l, m) + offset) then
+                    old_m = m
+                    old_l = l
+                    old_k = k 
+                   cycle
+                  else
+                    ! Target is at this lattice position
+                    lat_offset = lat % offset(map, k, l, m)
+                    offset = offset + lat_offset
+                    next_univ => universes(lat % universes(k, l, m))  
+                    path = trim(path) // "(" // trim(to_str(k)) // &
+                         "," // trim(to_str(l)) // "," // &
+                         trim(to_str(m)) // ")"
+                    call find_offset(map, goal, next_univ, final, offset, path)
+                    return
+                  end if
+               
+                end do
+              end do
+            end do
+
+          ! ==================================================================
+          ! HEXAGONAL LATTICES
+          type is (HexLattice)
+
+            ! Write to the geometry stack
+            path = trim(path) // "->" // to_str(lat%id)
+
+            n_z = lat % n_axial
+            n_y = 2 * lat % n_rings - 1
+            n_x = 2 * lat % n_rings - 1
+            old_m = 1
+            old_l = 1
+            old_k = 1
+
+            ! Loop over lattice coordinates
+            do m = 1, n_z
+              do l = 1, n_y
+                do k = 1, n_x
+
+                  ! This array position is never used
+                  if (k + l < lat % n_rings + 1) then
+                    cycle
+                  ! This array position is never used
+                  else if (k + l > 3*lat % n_rings - 1) then
+                    cycle
+                  end if
+
+                  if (final >= lat % offset(map, k, l, m) + offset) then
+                    old_m = m
+                    old_l = l
+                    old_k = k
+                    cycle
+                  else
+                    ! Target is at this lattice position
+                    lat_offset = lat % offset(map, old_k, old_l, old_m)
+                    offset = offset + lat_offset
+                    next_univ => universes(lat % universes(old_k, old_l, old_m))
+                    path = trim(path) // "(" // trim(to_str(old_k)) // &
+                         "," // trim(to_str(old_l)) // "," // &
+                         trim(to_str(old_m)) // ")"
```

Comment:
Yes, this seems reasonable enough to me. In reality, I'll go with whatever indexing scheme you think is best since you're the mastermind behind hex lattices. I simply was trying to figure out a way to print a semi-useful "path" for each distribcell to "tallies.out" here, so this is just the simplest scheme I knew how to do for hex lattices. I just revised as you suggested and pushed. You might try running "examples/python/lattice/hexagonal/build-xml.py" to see if the paths that are printed out are as you expect / wish.

Also, I really, really like your writeup on lattice indexing in OpenMC. Do you plan to submit that as a PR for the main website?